### PR TITLE
Update embedded appearance API with changes from API review

### PR DESCRIPTION
**Pull request by porter-stripe**
## Summary
- Moved EmbeddedPaymentElement.Style struct to EmbeddedPaymentElement.Row.Style
- Moved EmbeddedPaymentElement.style property to EmbeddedPaymentElement.row.style

## Motivation
- Embedded appearance API changes

## Testing
- Manual/CI

## Changelog
N/A

### CODE REVIEW

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheetAppearance.swift [hunk 1]
```diff
@@ -237,25 +237,25 @@ public extension PaymentSheet {
...
+            public enum Style: CaseIterable {
+                /// A flat style with radio buttons
+                case flatWithRadio
+                /// A floating button style
+                case floatingButton
+            }
+            
+            /// The display style of the Embedded Mobile Payment Element
+            public var style: Style = .flatWithRadio
+
```
Comment (wooj-stripe) [inline StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheetAppearance.swift:258]: The doc strings for these fields should be updated to match what is in the API review. E.g. for `var style: Style`, it should be:
```
            /// The display style of the row
```

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/STPAnalyticsClient+PaymentSheet.swift [hunk 2]
```diff
@@ -86,7 +86,7 @@ extension PaymentSheet.Appearance.EmbeddedPaymentElement {
 
     var analyticPayload: [String: Any] {
         var payload = [String: Any]()
-        payload["style"] = style != PaymentSheet.Appearance.EmbeddedPaymentElement.default.style
+        payload["style"] = row.style != PaymentSheet.Appearance.EmbeddedPaymentElement.default.row.style
         payload["row"] = row != PaymentSheet.Appearance.EmbeddedPaymentElement.default.row
```
Comment (wooj-stripe) [inline StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/STPAnalyticsClient+PaymentSheet.swift:90]: While we still have the chance (pre beta/GA), seems like we should update the key for this to `row_style`?

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheetAppearance.swift (hunk 1 above)
Comment (porter-stripe) [reply StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheetAppearance.swift]: Ah, didn't catch those were changed, thanks!

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/STPAnalyticsClient+PaymentSheet.swift (hunk 2 above)
Comment (wooj-stripe) [reply StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/STPAnalyticsClient+PaymentSheet.swift]: Hmm at some point in time, we should update our `PaymentSheetAnalyticsHelperTest` to include unit tests for this.

**Pull request merged**